src/anyio/_core/_fileio.py

```diff
@@ -514,8 +514,17 @@ async def read_text(
     ) -> str:
         return await to_thread.run_sync(self._path.read_text, encoding, errors)
 
-    def relative_to(self, *other: str | PathLike[str]) -> Path:
-        return Path(self._path.relative_to(*other))
+    if sys.version_info >= (3, 12):
+
+        def relative_to(
+            self, *other: str | PathLike[str], walk_up: bool = False
+        ) -> Path:
+            return Path(self._path.relative_to(*other, walk_up=walk_up))
```

Comment:
Add a test for this please.

Reply:
Added. Required a little bit of modification to the test environment to provide a sibling folder to try to reference. Let me know if this isn't acceptable - I can iterate on another approach